engine/runtime-language-arrow/src/main/java/org/enso/interpreter/arrow/runtime/ArrowFixedArrayInt.java

```diff
@@ -96,4 +128,133 @@ boolean isArrayElementReadable(long index) {
   private static int typeAdjustedIndex(long index, SizeInBytes unit) {
     return Math.toIntExact(index * unit.sizeInBytes());
   }
+
+  @ExportLibrary(InteropLibrary.class)
+  static final class LongIterator implements TruffleObject {
+    private int at;
+    private final ByteBuffer buffer;
+    @NeverDefault final int step;
+
+    LongIterator(ByteBuffer buffer, int step) {
+      assert step != 0;
+      this.buffer = buffer;
+      this.step = step;
+    }
+
+    @ExportMessage
+    Object getIteratorNextElement(
+        @Bind("$node") Node node,
+        @Cached("this.step") int step,
+        @Cached InlinedExactClassProfile bufferTypeProfile)
+        throws StopIterationException {
+      var buf = bufferTypeProfile.profile(node, buffer);
+      try {
+        var res = buf.getLong(at);
+        at += step;
+        return res;
+      } catch (BufferOverflowException ex) {
+        CompilerDirectives.transferToInterpreter();
+        throw StopIterationException.create();
+      }
+    }
```

Comment:
How does this logic relate to Arrow's data format?

I can see pure Java operations (`ByteBuffer::getLong`) here. Normally I'd expect some Arrow-related calls that encapsulate how Arrow is storing the data.

This makes me think we make some assumptions about Arrows' data layout. Or am I misunderstanding something?

Reply:
This is an iterator for `Int64` Arrow type. Luckily the type exactly matches `long`. As such we just `getLong` here and that's it.

Reply:
I see. Is it worth documenting somewhere that we are relying on the data layout being like that?

Reply:
The last time I asked for a documentation, I was told: _the code is the documentation_...

> documenting somewhere

...if so, and until better documentation is written, we can consider the `Int64` == `long` to be documented.